### DISCUSSION TITLE
self hosted git repository with self signed certificate

### DISCUSSION THREAD

**Discussion by Masber** [GitoxideLabs/gitoxide#1305] [Q&A]
Dear gitoxide community,

I am building a cli in Rust which interacts with a self hosted git repository supported by gitea (https://about.gitea.com/). One of the functionalities of this cli is to show refs (tag and branch name), author name/email, committer name/email and commit date and time for a specific sha value from the remote repo.

My problem is that this remote repo has a self signed certificate.

My questions are:

1. How can I inject my self hosted/local git server certificate into gitoxide?
2. Do I need to create a local git repo (folder) in order to fetch from external repos if I only want to get some information and print them on screen? can I do this in memory without having to handle silesystem io and or files/folders?

thank you very much

**Comment by Byron** [comment]
That's a great problem to tackle!

To your questions.

#### How can I inject my server certificate into gitoxide?

One can set the git configuration `http.sslVerify=false` and it won't validate the certificate anymore. Ideally, this is only set where needed to get full validation otherwise (and MITM protection).

#### Do I need to create a local git repo (folder) in order to fetch from repo and process the information in memory?

Yes, `gitoxide` is very much tried to a filesystem at the moment, so real clones and fetches would have to happen before any analysis on that data can be done. Besides that, you should already have all the tools you need to build what you want.

All the best!

**Reply by Masber** [reply]
ok, thanks for the prompt response!

is there any plan to support server cert injection in the near future?

thanks!

**Reply by Byron** [reply]
You can typically use anything from the [available Git configuration](https://git-scm.com/docs/git-config), and use `gix config-tree` to see if configuration key is already supported.
